ci/setup-build-test-gdk.ps1

```diff
@@ -3,25 +3,38 @@ param(
     [string] $gcs_publish_bucket = "io-internal-infra-unreal-artifacts-production/UnrealEngine",
     [string] $msbuild_exe = "${env:ProgramFiles(x86)}\Microsoft Visual Studio\2019\BuildTools\MSBuild\Current\Bin\MSBuild.exe",
     [string] $build_home = (Get-Item "$($PSScriptRoot)").parent.parent.FullName, ## The root of the entire build. Should ultimately resolve to "C:\b\<number>\".
-    [string] $unreal_engine_symlink_dir = "$build_home\UnrealEngine"
+    [string] $unreal_engine_symlink_dir = "$build_home\UnrealEngine",
+    [string] $gym_version_path = "$unreal_engine_symlink_dir\UnrealGDKTestGymVersion.txt"
 )
 
 class TestProjectTarget {
     [ValidateNotNullOrEmpty()][string]$test_repo_url
     [ValidateNotNullOrEmpty()][string]$test_repo_branch
     [ValidateNotNullOrEmpty()][string]$test_repo_relative_uproject_path
     [ValidateNotNullOrEmpty()][string]$test_project_name
+    [ValidateNotNullOrEmpty()][string]$test_gym_version_path
+    [ValidateNotNull()][string]$test_env_override
```

Comment:
```suggestion
    [ValidateNotNull()][string]$test_version_override
```

Or maybe this should be outside of TestProjectTarget's constructor, so a correct git ref is passed in? Idk, passing both a version file path and an override version looks somewhat strange to me.

Reply:
I want all the override logic in one place so its more explicit. If I move the logic of deciding the correct ref out of the function as you suggest, we end up with split logic as before. Its maybe a bit verbose and odd but at least there is one place in which I can see the override structure.

Its all a bit convoluted at the moment but I don't want to change how its been working for a while. As mentioned above the hierarchy we seem to have settled on env var > branch name > text file > master and I am keen to keep that in place for now.